### PR TITLE
[citrix_adc] Add support of citrix_waf connector and enhance the syslog message

### PR DESCRIPTION
**Pull request by janvi-elastic**
## What does this PR do?

- Added log data stream.
- Added data collection logic for log data stream.
- Added the ingest pipeline for log data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added dashboards and visualizations.
- Added test for pipeline for log data stream.
- Added system test cases for log data stream.

## Integration release checklist
This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes
- [x] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md)
- [x] Supported versions of the monitoring target is documented
- [x] Supported operating systems are documented (if applicable)
- [x] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [x] Documentation exists
- [x] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [x] At least a manual test with ES / Kibana / Agent has been performed.
- [x] Required Kibana version set to: ^8.12.0

### New Package
- [x] Screenshot of the "Add Integration" page on Fleet added

### Dashboards changes
- [x] Dashboards exists
- [x] Screenshots added or updated
- [x] Datastream filters added to visualizations

### Log dataset changes
- [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [x] Generated output for at least 1 log file exists
- [x] Sample event (sample_event.json) exists

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/citrix_adc directory.
- Run the following command to run tests. 
> elastic-package test 

## Related issues

- Closes https://github.com/elastic/integrations/issues/470

## Automated Test
```
--- Test results for package: citrix_adc - START ---
╭────────────┬─────────────┬───────────┬──────────────────────────┬────────┬──────────────╮
│ PACKAGE    │ DATA STREAM │ TEST TYPE │ TEST NAME                │ RESULT │ TIME ELAPSED │
├────────────┼─────────────┼───────────┼──────────────────────────┼────────┼──────────────┤
│ citrix_adc │ interface   │ static    │ Verify sample_event.json │ PASS   │  92.279815ms │
│ citrix_adc │ lbvserver   │ static    │ Verify sample_event.json │ PASS   │ 103.131254ms │
│ citrix_adc │ log         │ static    │ Verify sample_event.json │ PASS   │ 148.748599ms │
│ citrix_adc │ service     │ static    │ Verify sample_event.json │ PASS   │  99.383878ms │
│ citrix_adc │ system      │ static    │ Verify sample_event.json │ PASS   │  101.77641ms │
│ citrix_adc │ vpn         │ static    │ Verify sample_event.json │ PASS   │  99.276704ms │
╰────────────┴─────────────┴───────────┴──────────────────────────┴────────┴──────────────╯
--- Test results for package: citrix_adc - END   ---
Done
--- Test results for package: citrix_adc - START ---
╭────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE    │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ citrix_adc │ interface   │ system    │ default   │ PASS   │ 35.164726297s │
│ citrix_adc │ lbvserver   │ system    │ default   │ PASS   │ 32.908577723s │
│ citrix_adc │ log         │ system    │ logfile   │ PASS   │ 33.045892741s │
│ citrix_adc │ log         │ system    │ tcp       │ PASS   │ 34.854942372s │
│ citrix_adc │ log         │ system    │ tls       │ PASS   │ 34.738035093s │
│ citrix_adc │ log         │ system    │ udp       │ PASS   │ 35.646803319s │
│ citrix_adc │ service     │ system    │ default   │ PASS   │ 33.613070758s │
│ citrix_adc │ system      │ system    │ default   │ PASS   │ 32.208498886s │
│ citrix_adc │ vpn         │ system    │ default   │ PASS   │ 31.249781627s │
╰────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: citrix_adc - END   ---
Done
--- Test results for package: citrix_adc - START ---
╭────────────┬─────────────┬───────────┬─────────────────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE    │ DATA STREAM │ TEST TYPE │ TEST NAME                                                           │ RESULT │ TIME ELAPSED │
├────────────┼─────────────┼───────────┼─────────────────────────────────────────────────────────────────────┼────────┼──────────────┤
│ citrix_adc │             │ asset     │ dashboard citrix_adc-2b30a8f0-4fa9-11ed-8fa7-7bab33159b99 is loaded │ PASS   │      1.473µs │
│ citrix_adc │             │ asset     │ dashboard citrix_adc-73ef1be0-485a-11ed-aee6-31b55c85e6df is loaded │ PASS   │        184ns │
│ citrix_adc │             │ asset     │ dashboard citrix_adc-8d0661f0-4fa4-11ed-8fa7-7bab33159b99 is loaded │ PASS   │        169ns │
│ citrix_adc │             │ asset     │ dashboard citrix_adc-95709fd0-e130-11ee-adb0-b71252739438 is loaded │ PASS   │        162ns │
│ citrix_adc │             │ asset     │ dashboard citrix_adc-abcd5660-4947-11ed-9b28-1f7d06bfd481 is loaded │ PASS   │        159ns │
│ citrix_adc │             │ asset     │ dashboard citrix_adc-b475f280-4eb1-11ed-9db6-73aea65de09b is loaded │ PASS   │        217ns │
│ citrix_adc │             │ asset     │ dashboard citrix_adc-c4b9b970-3d99-11ed-9f8b-1bc5a55dfeec is loaded │ PASS   │        241ns │
│ citrix_adc │ interface   │ asset     │ index_template logs-citrix_adc.interface is loaded                  │ PASS   │        551ns │
│ citrix_adc │ interface   │ asset     │ ingest_pipeline logs-citrix_adc.interface-1.4.0 is loaded           │ PASS   │        209ns │
│ citrix_adc │ lbvserver   │ asset     │ index_template logs-citrix_adc.lbvserver is loaded                  │ PASS   │        354ns │
│ citrix_adc │ lbvserver   │ asset     │ ingest_pipeline logs-citrix_adc.lbvserver-1.4.0 is loaded           │ PASS   │        204ns │
│ citrix_adc │ log         │ asset     │ index_template logs-citrix_adc.log is loaded                        │ PASS   │        276ns │
│ citrix_adc │ log         │ asset     │ ingest_pipeline logs-citrix_adc.log-1.4.0 is loaded                 │ PASS   │        170ns │
│ citrix_adc │ service     │ asset     │ index_template logs-citrix_adc.service is loaded                    │ PASS   │        227ns │
│ citrix_adc │ service     │ asset     │ ingest_pipeline logs-citrix_adc.service-1.4.0 is loaded             │ PASS   │        396ns │
│ citrix_adc │ system      │ asset     │ index_template logs-citrix_adc.system is loaded                     │ PASS   │        290ns │
│ citrix_adc │ system      │ asset     │ ingest_pipeline logs-citrix_adc.system-1.4.0 is loaded              │ PASS   │        304ns │
│ citrix_adc │ vpn         │ asset     │ index_template logs-citrix_adc.vpn is loaded                        │ PASS   │        386ns │
│ citrix_adc │ vpn         │ asset     │ ingest_pipeline logs-citrix_adc.vpn-1.4.0 is loaded                 │ PASS   │        372ns │
╰────────────┴─────────────┴───────────┴─────────────────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: citrix_adc - END   ---
Done
--- Test results for package: citrix_adc - START ---
╭────────────┬─────────────┬───────────┬────────────────────────────┬────────┬──────────────╮
│ PACKAGE    │ DATA STREAM │ TEST TYPE │ TEST NAME                  │ RESULT │ TIME ELAPSED │
├────────────┼─────────────┼───────────┼────────────────────────────┼────────┼──────────────┤
│ citrix_adc │ interface   │ pipeline  │ test-interface-metrics.log │ PASS   │   7.614029ms │
│ citrix_adc │ lbvserver   │ pipeline  │ test-lbvserver-metrics.log │ PASS   │    3.14102ms │
│ citrix_adc │ log         │ pipeline  │ test-citrix-waf-cef.log    │ PASS   │  11.488881ms │
│ citrix_adc │ log         │ pipeline  │ test-citrix-waf-native.log │ PASS   │ 105.808545ms │
│ citrix_adc │ service     │ pipeline  │ test-service-metrics.log   │ PASS   │   3.007045ms │
│ citrix_adc │ system      │ pipeline  │ test-system-metrics.log    │ PASS   │    3.17727ms │
│ citrix_adc │ vpn         │ pipeline  │ test-vpn-metrics.log       │ PASS   │   2.314888ms │
╰────────────┴─────────────┴───────────┴────────────────────────────┴────────┴──────────────╯
--- Test results for package: citrix_adc - END   ---
Done
```

## Screenshot
image
image

### CODE REVIEW

**Comment by muthu-mps** [comment]
@janvi-elastic
- As this datastream adds support for `Citrix WAF` connector in the Citrix ADC integration and collect the logs.
-  I do see that the integration to support `Citrix WAF` exists already and performs collecting the log formats processed in this datastream. 
    - Reference: https://github.com/elastic/integrations/tree/main/packages/citrix_waf#:~:text=Logs-,Syslog,-The%20citrix_waf.log.
- This is also tested with the Citrix ADC logs as mentioned [here](https://github.com/elastic/integrations/tree/main/packages/citrix_waf#:~:text=samples%20obtained%20from-,Citrix%20ADC%2013.1,-and%20NetScaler%2010.0).
- Can we use the existing WAF integration to collect the CEF log formats?

**Comment by jamiehynds** [comment]
@muthu-mps our initial intent was to add support for WAF events via the Netscaler/ADC integration, and eventually deprecate the standalone WAF integration to avoid having 2 integrations for the same product. However, after looking further into Citrix docs, the WAF is available as a stand-alone appliance, or as a feature on a NetScaler virtual appliance so it possibly makes sense to keep WAF as a standalone integration and not merging into Netscaler. 

@efd6 can you recall if the WAF integration you built was based on a standalone WAF or a feature of Netscaler? I'm presuming the eventsa are the same across both.

**Comment by efd6** [comment]
From memory, the integrations was build against events in the documentation [here](https://docs.netscaler.com/en-us/citrix-adc/current-release/application-firewall/logs.html). This is confirmed by comparing the test events with the messages listed in the documentation.

**Comment by kcreddy** [comment]
> However, after looking further into Citrix docs, the WAF is available as a stand-alone appliance, or as a feature on a NetScaler virtual appliance so it possibly makes sense to keep WAF as a standalone integration and not merging into Netscaler.

@jamiehynds It looks like we are in favour of moving this support into Citrix WAF integration (if doesn't exist already). Can you please confirm?

**Comment by jamiehynds** [comment]
> > However, after looking further into Citrix docs, the WAF is available as a stand-alone appliance, or as a feature on a NetScaler virtual appliance so it possibly makes sense to keep WAF as a standalone integration and not merging into Netscaler.
> 
> @jamiehynds It looks like we are in favour of moving this support into Citrix WAF integration (if doesn't exist already). Can you please confirm?

Given that WAF is available as a standalone appliance, it makes sense to keep it as it's own integration. This also ensures no impact to existing users of the integration. 

The original plan was to include support for WAF events into the ADC integration also, as WAF can also run as part of ADC. Is it possible to add the WAF pipeline to the ADC integration, so users running WAF as part of ADC just have to enable a single integration? Then users who are running WAF standalone, can run the WAF integration? The alternative would be having the ADC integration without WAF support - ADC users would then have to run both the ADC and WAF integration.

**Comment by kcreddy** [comment]
> The goal is to keep the existing WAF integration in place for users running WAF standalone, and include the WAF pipelines as part of the ADC integration for those running WAF as part of ADC


Thanks for the clarifications Jamie.

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

File: packages/citrix_adc/data_stream/log/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,238 @@
+title: Cisco ADC logs
```
Comment (muthu-mps) [inline packages/citrix_adc/data_stream/log/manifest.yml:1]:
```suggestion
title: Citrix ADC logs
```

**Comment by muthu-mps** [comment]
- Can we check the dashboard image and remove if its not associated with this datastream?
      Reference: `packages/citrix_adc/img/dashboard.png`
- Just to check, Is it possible to apply the `data_stream.dataset `filter at dashboard level?

**Comment by eriroley** [comment]
> > > However, after looking further into Citrix docs, the WAF is available as a stand-alone appliance, or as a feature on a NetScaler virtual appliance so it possibly makes sense to keep WAF as a standalone integration and not merging into Netscaler.
> > 
> > 
> > @jamiehynds It looks like we are in favour of moving this support into Citrix WAF integration (if doesn't exist already). Can you please confirm?
> 
> Given that WAF is available as a standalone appliance, it makes sense to keep it as it's own integration. This also ensures no impact to existing users of the integration. The goal is to keep the existing WAF integration in place for users running WAF standalone, and include the WAF pipelines as part of the ADC integration for those running WAF as part of ADC.

My reading of this is that the WAF integration would then not be needed (would be redundant) for folks running WAF as part of ADC - if that is correct, that should be included as a note/large block in the integration information

**Comment by jamiehynds** [comment]
> > > > However, after looking further into Citrix docs, the WAF is available as a stand-alone appliance, or as a feature on a NetScaler virtual appliance so it possibly makes sense to keep WAF as a standalone integration and not merging into Netscaler.
> > > 
> > > 
> > > @jamiehynds It looks like we are in favour of moving this support into Citrix WAF integration (if doesn't exist already). Can you please confirm?
> > 
> > 
> > Given that WAF is available as a standalone appliance, it makes sense to keep it as it's own integration. This also ensures no impact to existing users of the integration. The goal is to keep the existing WAF integration in place for users running WAF standalone, and include the WAF pipelines as part of the ADC integration for those running WAF as part of ADC.
> 
> My reading of this is that the WAF integration would then not be needed (would be redundant) for folks running WAF as part of ADC - if that is correct, that should be included as a note/large block in the integration information

Great point, thanks @eriroley. I'll create an issue to ensure this is noted in the WAF docs.

**Comment by janvi-elastic** [comment]
> * Can we check the dashboard image and remove if its not associated with this datastream?
>   Reference: `packages/citrix_adc/img/dashboard.png`
> * Just to check, Is it possible to apply the `data_stream.dataset `filter at dashboard level?

@muthu-mps We follow the best practice of adding `data_stream.dataset` filter at the visualization level instead of dashboard level in Security Connectors. Please LMK your thoughts.

**Comment by muthu-mps** [comment]
> > * Can we check the dashboard image and remove if its not associated with this datastream?
> >   Reference: `packages/citrix_adc/img/dashboard.png`
> > * Just to check, Is it possible to apply the `data_stream.dataset `filter at dashboard level?
> 
> @muthu-mps We follow the best practice of adding `data_stream.dataset` filter at the visualization level instead of dashboard level in Security Connectors. Please LMK your thoughts.

This is fine, As its followed as best practise across integrations.

**Review by muthu-mps — APPROVED** [review]
LGTM!

**Comment by jamiehynds** [comment]
@janvi-elastic sorry, I just took at look at the 1.5 package now in production. One minor nit. Can we create a PR to adjust the package description to convey that the integration now supports logs. i.e. `This integration collects metrics and logs from Citrix ADC`

**Comment by piyush-elastic** [comment]
@jamiehynds - I could see we have added Note in Readme [here](https://github.com/elastic/integrations/pull/9385/files#diff-1eb9db55c17000b88f1831de49d97aede088119068f67c4b7a6057f012e11065R26). Let me know if you are expecting same.

**Comment by jamiehynds** [comment]
@piyush-elastic it's actually here that we need to adjust, to ensure the description makes it clear that logs are supported, and not just metrics. https://github.com/elastic/integrations/pull/9385/files#diff-fdd22477409b9bbe63be0153c9250192f8aea9b59e16aa28f0d60d17b58bb049L5

**Pull request merged**